src/pykx/pandas_api/pandas_meta.py

```diff
@@ -154,6 +155,29 @@ def mean(self, axis: int = 0, numeric_only: bool = False):
             tab
         )
 
+    @api_return
+    def std(self, axis: int = 0, ddof: int = 1, numeric_only: bool = False):
+        tab = self
+        if 'Keyed' in str(type(tab)):
+            tab = q('{(keys x) _ 0!x}', tab)
+        if numeric_only:
+            tab = _get_numeric_only_subtable(tab)
+        key_str = '' if axis == 0 else '`$string '
+        val_str = '' if axis == 0 else '"f"$value '
+        query_str = 'cols[tab]' if axis == 0 else 'til[count[tab]]'
+        where_str = ' where not (::)~/:r[;1]'
+
+        res = q(f'{{[tab]{query_str}!count[{query_str}]#0n}}', tab)
+        if ddof != len(tab.pd()):
+            res = q(
+                '{[tab]'
+                f'r:{{[tab; x] ({key_str}x; {{avg sqrt (sum xexp[x-(avg x);2]) % count[x]-{ddof}}} {val_str}tab[x])}}[tab;] each {query_str};'
```

Comment:
This line is too long, we should shorten it. Would it be possible to leverage `dev` and `sdev` for the cases where `ddof` values `0` and `1`, respectively? A similar pattern to the one that we see at `key_str` would serve us.

Reply:
I haven't run it, so it probably has (syntax) errors, but what about the following implementation for the `std` function? It leverages `dev` and `sdev`, it avoids calling `pd` to check the count of elements and it shortens the large line into more digestible chunks. What do you think?
```python
@api_return
def std(self, axis: int = 0, ddof: int = 1, numeric_only: bool = False):
    tab = self
    if 'Keyed' in str(type(tab)):
        tab = q('{(keys x) _ 0!x}', tab)
    if numeric_only:
        tab = _get_numeric_only_subtable(tab)

    key_str = '' if axis == 0 else '`$string '
    val_str = '' if axis == 0 else '"f"$value '
    query_str = 'cols[tab]' if axis == 0 else 'til[count[tab]]'
    where_str = ' where not (::)~/:r[;1]'
    x_dev_str = f'{{avg sqrt (sum xexp[x-avg x;2]) % count[x]-{ddof}}}'
    dev_str = 'dev ' if ddof == 0 else 'sdev ' if ddof == 1 else x_dev_str

    if ddof == q('count', tab):
        return q(f'{{[tab]{query_str}!count[{query_str}]#0n}}', tab)

    return q(
        '{[tab]'
        f'r:{{[tab; x] ({key_str}x; {dev_str} {val_str}tab[x])}}[tab;] each {query_str};'
        f'(,/) {{(enlist x 0)!(enlist x 1)}} each r{where_str}}}',
        tab
    )
```

Reply:
Overall it looks good to me and it passes all the tests we designed. It looks like `len(tab)` would also work as a way to count the rows on a PyKX Table. Using this we could avoid summoning q code if it's not necessary, and it doesn't affect the results of the tests. What do you think?

```python
if ddof == len(tab):
    return q(f'{{[tab]{query_str}!count[{query_str}]#0n}}', tab)
```

Reply:
As we discussed on a call earlier, I think the last part of the function could be a bit confusing since there are two `return` calls. Even if correct, I have not seen it elsewhere on the library, so my proposal looks like this:

```python
    @api_return
    def std(self, axis: int = 0, ddof: int = 1, numeric_only: bool = False):
        tab = self
        if 'Keyed' in str(type(tab)):
            tab = q('{(keys x) _ 0!x}', tab)
        if numeric_only:
            tab = _get_numeric_only_subtable(tab)

        key_str = '' if axis == 0 else '`$string '
        val_str = '' if axis == 0 else '"f"$value '
        query_str = 'cols[tab]' if axis == 0 else 'til[count[tab]]'
        where_str = ' where not (::)~/:r[;1]'
        x_dev_str = f'{{avg sqrt (sum xexp[x-avg x;2]) % count[x]-{ddof}}}'
        dev_str = 'dev' if ddof == 0 else 'sdev' if ddof == 1 else x_dev_str
        null_str = f'{{[tab]{query_str}!count[{query_str}]#0n}}'
        regular_ops_str = '{[tab]' + \
            f'r:{{[tab; x] ({key_str}x; {dev_str} {val_str}tab[x])}}[tab;] each {query_str};' + \
            f'(,/) {{(enlist x 0)!(enlist x 1)}} each r{where_str}}}'

        return q(null_str if ddof == len(tab) else regular_ops, tab)
```

Not sure if I like it or not, since I needed to add a ` + \` at the end of each part of the `regular_ops_str` string, but I think that removing the `+` while keeping the `\` would do the trick. What do you think?

Reply:
Visually, I prefer my version, since the connection with `mean` and `median` is more direct and it avoids the `+\` at the end. As I mentioned offline, the `null` return could be seen as an exceptional situation, and therefore I find the style all right. Undoubtedly, this is more of a personal taste, so feel free to choose whichever version you like the most! Anyway, beware of the variable names, you are returning `regular_ops` at the end, but such variable should be `regular_ops_str` instead (btw. what about a shorter name here like `regular_str` or even `std_str`? ). Thanks!